### PR TITLE
Use vendor go mod flag on all calls to `go install`

### PR DESCRIPTION
**Pull request by ysimonson**
Otherwise pachyderm will build from the normal go mod directory, rather than our vendored directory.

### CODE REVIEW

**Comment by ysimonson** [comment]
[`mod=vendor`](https://github.com/pachyderm/pachyderm/pull/4134/commits/31bf0fc8500d2d5ec51581fef744f9ad813c28da) was removed for a reason, so instead experimenting with entirely removing vendored libs: https://github.com/pachyderm/pachyderm/pull/4323

**Comment by ysimonson** [comment]
@msteffen I'm still working on trying to remove vendoring all-together in a separate PR, but it's a big uphill battle. In the mean-time, this change has the merit of working. Maybe it makes sense to consider this change (at the very least as a stopgap to ensure CI and local dev act the same way), and get the vendoring removal merged when/if it's working. What do you think?

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ cache:
     # etc/testing/travis_cache.sh
     - ${HOME}/.cache
     - ${HOME}/cached-deps
+    - ${HOME}/gopath/pkg/mod
```
Comment (msteffen) [inline .travis.yml]: Would it make sense to use `$GOPATH` instead of `$HOME/gopath`?

File: etc/testing/travis_cache.sh [hunk 2]
```diff
@@ -11,6 +11,7 @@ set -ex
 dirs=(
   "${HOME}/.cache"
   "${HOME}/cached-deps"
+  "${HOME}/gopath/pkg/mod"
```
Comment (msteffen) [inline etc/testing/travis_cache.sh]: Similar to above: would it make sense to use `$GOPATH` instead of `$HOME/gopath`?

**Comment by msteffen** [comment]
Sure! To be clear, my personal opinion is that I regard this to be equally good to `GO111MODULES=off` (I argued for switching b/c I was having trouble with `make lint`, but it looks like adding `$HOME/gopath/pkg/mod/cache` to `travis_cache.sh` fixed that), so if you think it's better, then I think we should do it. I just think if we *could* get rid of vendor, then that would be the strict improvement

One high-level question: do we want to add `-mod=vendor` to `go build` everywhere as well? My understanding is that otherwise we won't use the `vendor` dir for any `go build` commands (the PR where I switched it is here: https://github.com/pachyderm/pachyderm/pull/4134/files, so that could also be useful if you're trying to find all the places to switch back)

**Comment by ysimonson** [comment]
Yeah that makes sense, made the change

**Comment by ysimonson** [comment]
@msteffen I tried caching the entire gopath, but that leads to a problem: TravisCI clones the repo into $GOPATH, but also tries to pull the directory from cache. This leads to permissioning issues.

**Comment by ysimonson** [comment]
Closing this out in favor of #4323

**Pull request closed without merging**